### ISSUE TITLE
[Dependency] Kerbal Atomic

### ISSUE THREAD

**Issue by yalov** [KSP-CKAN/NetKAN#7400]
`Kerbal Atomics - Liquid Fuel Only` cancels dependent of `CryoTanks` for the `Kerbal Atomic`, correct? 
Hovewer, CKAN forces me to install `CryoTanks`, even if I am planning to use `Kerbal Atomics - Liquid Fuel Only`

@ChrisAdderley

**Comment by ChrisAdderley** [comment]
Eh, It's kinda a fairly unsupported configuration. I'd have to get some clarity from CKAN people on how to resolve this loop:
* KA requires CT
* KALF requires KA
* KALF technically doesn't require CT.

**Comment by HebaruSan** [comment]
If I'm understanding this correctly (probably not), you could add this to `KerbalAtomics-NTRsUseLF`:

```json
    "provides": [ "CryoTanks" ],
```

That way, installing `KerbalAtomics-NTRsUseLF` would "count" as installing `CryoTanks` without actually installing it. Is that what we're trying to do?

**Comment by ChrisAdderley** [comment]
So if you installed `KerbalAtomics-NTRsUseLF`, it would require `KerbalAtomics`, which would need to satisfy its `CryoTanks` requirements. You're saying it that it could count as satisfied by adding this new option.

Are there negative things that are then going to happen (eg. now `KerbalAtomics-NTRsUseLF` can now satisfy the requirements of other mods that need CryoTanks)?

**Comment by HebaruSan** [comment]
Hmm, yes, if you install another mod that depends on `CryoTanks`, you would be prompted to choose between `CryoTanks` and `KerbalAtomics-NTRsUseLF` to satisfy the dependency. I suppose that's not ideal.

What if `KerbalAtomics` instead did this:

```json
    "depends": [
        { "any_of": [
            { "name": "CryoTanks" },
            { "name": "KerbalAtomics-NTRsUseLF" }
        ] }
    ],
```

That way, any other module's explicit dependency on `CryoTanks` would be unaffected, but `KerbalAtomics` would only require one or the other. Is that closer?

**Comment by ChrisAdderley** [comment]
That seems a lot closer. I will put it on the list to evaluate and test, and close this issue when it is implemented.

**Comment by yalov** [comment]
btw, configs for Fuel Switching for stock tanks is really what necessary for the KA, not the whole CT. CT is more like recommends. So maybe the best way is
```
    "depends": [
        { "any_of": [
            { "name": "FSLqdHydrogenPatch" },
            { "name": "KerbalAtomics-NTRsUseLF" }
        ] }
    ],
```
It's the same for the CryoEngines. 
Only problem is that user can miss CT, if KA and CE is installing as recommends of another mod. So maybe add CT as third option to depends any_of.

**Comment by ChrisAdderley** [comment]
No, it's needed. Although in a highly technical sense, one could go through all the other mods that might provide LH2 tank switching and add them as usable alternates, the reality is that they all have inconsistent and sometimes completely broken mass ratios and are therefore only surficial alternates.

**Comment by HebaruSan** [comment]
There's a syntax error in the new netkan, fix here: ChrisAdderley/KerbalAtomics#81

**Comment by HebaruSan** [comment]
Installing KerbalAtomics now prompts to choose CryoTanks or LF Only (unless you already have one of them installed):

![image](https://user-images.githubusercontent.com/1559108/68452180-4070d200-01b7-11ea-8d3a-8408db0cc1b3.png)
